### PR TITLE
[fix] [doc] fix multiple apis in the automatically generated documentation use the same anchor point

### PR DESCRIPTION
**Pull request by poorbarcode**
### Motivation

Now that site [admin-rest-api](https://pulsar.apache.org/admin-rest-api) is automatically generated through swagger, each API in the site has its own link that we can walk through, the generated rule of API-link is `https://pulsar.apache.org/admin-rest-api/#operation/{operationId}`, we call the suffix `#operation/{operationId}` of link anchor.

Since the {operationId}'s default value is the method name of the endpoint class, and there has the same method name in the multi endpoint-classes, such as `Namespaces.setDispatchRate` and `PersistentTopics.setDispatchRate`, so these both APIs will generate the same link: `https://pulsar.apache.org/admin-rest-api/#operation/setDispatchRate`.

### Modifications

In the update to version `3.1.8` of `swagger-maven-plugin,` there are another ways to set `operationId`: The Operation Id Format. see https://github.com/kongchen/swagger-maven-plugin/blob/e4cf3af6ed4a4987b2269a05b7037549bc4a5511/src/main/java/com/github/kongchen/swagger/docgen/reader/AbstractReader.java#L535-L540. So we can generate `operationId` using the specified rule `{className}_{methodName}.`

截屏2023-03-08 16 04 47

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [x] Anything that affects deployment

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: 
- https://github.com/poorbarcode/pulsar/pull/51

### CODE REVIEW

**Review by dave2wave — COMMENTED** [review]
These look like actual API changes. Please explain.

**Comment by poorbarcode** [comment]
Hi @dave2wave 

> These look like actual API changes. Please explain.

Already changed the plain to only modify Swagger annotations. please take a look again

**Comment by mattisonchao** [comment]
@tisonkun  What do you think about this PR? I am not familiar with swagger, but I know you have worked on this part recently.

**Comment by tisonkun** [comment]
I agree that it's not a breaking change for end-users but only affects the doc experience.

Some related works previous: 

* https://github.com/apache/pulsar/pull/19025
* https://github.com/apache/pulsar/pull/19065

And I agree that a nickname alias can be a better solution. I can approve this move but I appreciate seeing the discussion on the mailing list continue for a while (days or after the lunar new year holiday).

cc @tuteng

**Comment by Shoothzj** [comment]
Now we aggregate rest api into a single file. I think it might be better to split them. This is also good for us to generate other language admin clients. I am trying to work on it.

**Comment by poorbarcode** [comment]
@tisonkun @tuteng @heesung-sn 

Please take a look, thanks

File: pulsar-broker/pom.xml [hunk 1]
```diff
@@ -697,11 +697,13 @@
...
             <groupId>com.github.kongchen</groupId>
             <artifactId>swagger-maven-plugin</artifactId>
-            <version>3.1.7</version>
+            <version>3.1.8</version>
             <configuration>
               <apiSources>
                 <apiSource>
                   <springmvc>false</springmvc>
+                  <operationIdFormat>{{className}}_{{methodName}}</operationIdFormat>
+                  <outputFormats>json</outputFormats>
```
Comment (heesung-sn) [inline pulsar-broker/pom.xml:706]: Is this json outputFormats necessary?

File: pulsar-broker/pom.xml [hunk 2]
```diff
@@ -697,11 +697,13 @@
           <plugin>
             <groupId>com.github.kongchen</groupId>
             <artifactId>swagger-maven-plugin</artifactId>
-            <version>3.1.7</version>
+            <version>3.1.8</version>
             <configuration>
               <apiSources>
                 <apiSource>
                   <springmvc>false</springmvc>
+                  <operationIdFormat>{{className}}_{{methodName}}</operationIdFormat>
```
Comment (heesung-sn) [inline pulsar-broker/pom.xml:705]: I think `{className}_{methodName}` is reasonable as the unique operationIdFormat. 

LGTM if we don't have duplicated apiSources from the same class and methodName.

**Review by heesung-sn — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
Please take a look at the OWASP failure. Although it seems unrelated to this patch, we should fix it later.

File: pulsar-broker/pom.xml (hunk 1 above)
Comment (poorbarcode) [reply pulsar-broker/pom.xml:706]: > nit: Is this json outputFormats necessary?

It should be optional, but maven or swagger had issues, and I had null Pointers when testing locally. I tried troubleshooting the issue yesterday but couldn't get the root cause quickly, so I thought adding this configuration would avoid unnecessary trouble.

File: pulsar-broker/pom.xml (hunk 2 above)
Comment (poorbarcode) [reply pulsar-broker/pom.xml:705]: After keeping only the latest version of admin API(such as `v2`, `v3`), `{className}_{methodName}` will not conflict. If we open multiple versions of API in the future, we need to change the rule to `{package_name}_{className}_{methodName}`

**Comment by poorbarcode** [comment]
Hi @tisonkun 

> Please take a look at the OWASP failure. Although it seems unrelated to this patch, we should fix it later.

I have pushed a https://github.com/apache/pulsar/pull/19764 to fix it, so I am merging the current PR now.

**Pull request merged**